lib/graphql/language/generation.rb

```diff
@@ -53,32 +55,73 @@ def self.generate(node, indent: "")
           "#{node.name}"
         when Nodes::VariableDefinition
           out = "$#{node.name}: #{generate(node.type)}"
-          out << " = #{generate(node.default_value)}" if node.default_value
+          out << " = #{generate(node.default_value)}" unless node.default_value.nil?
           out
         when Nodes::VariableIdentifier
           "$#{node.name}"
+        when Nodes::SchemaDefinition
+          out = "schema {\n"
+          out << "  query: #{node.query}\n" if node.query
+          out << "  mutation: #{node.mutation}\n" if node.mutation
+          out << "  subscription: #{node.subscription}\n" if node.subscription
+          out << "}"
+        when Nodes::ScalarTypeDefinition
+          "scalar #{node.name}"
+        when Nodes::ObjectTypeDefinition
+          out = "type #{node.name}"
+          out << " implements " << node.interfaces.join(", ") unless node.interfaces.empty?
+          out << generate_field_definitions(node.fields)
+        when Nodes::InputValueDefinition
+          out = "#{node.name}: #{generate(node.type)}"
+          out << " = #{generate(node.default_value)}" unless node.default_value.nil?
+          out
+        when Nodes::FieldDefinition
+          out = node.name
```

Comment:
ruh roh!  I think we need a new string here, since it's being mutated below. Otherwise ... chaos!

```ruby
schema_doc = GraphQL.parse("type Mutation { post: Post }")
puts GraphQL::Language::Generation.generate(schema_doc)
# type Mutation {
#   post: Post
# }
puts GraphQL::Language::Generation.generate(schema_doc)
# type Mutation {
#   post: Post: Post
# }
puts GraphQL::Language::Generation.generate(schema_doc)
# type Mutation {
#   post: Post: Post: Post
# }
puts GraphQL::Language::Generation.generate(schema_doc)
# type Mutation {
#   post: Post: Post: Post: Post
# }
```